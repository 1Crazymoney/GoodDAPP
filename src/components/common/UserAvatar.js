import React from 'react'
import { StyleSheet, View } from 'react-native'
import Avatar from './Avatar'
import Section from './Section'

const UserAvatar = props => {
  const { profile } = props
  return (
    <View style={styles.avatar}>
      <View style={styles.innerAvatar}>
        <Avatar size={120} />
        <Section.Title>{profile.fullName}</Section.Title>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  avatar: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row'
  },
  innerAvatar: {
    flexDirection: 'column'
  }
})

export default UserAvatar
