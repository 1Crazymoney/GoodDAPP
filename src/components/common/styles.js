import { normalize } from 'react-native-elements'

export const fontStyle = {
  color: '#555',
  fontSize: normalize(18),
  textAlign: 'center',
  fontFamily: 'Roboto'
}

export const scrollableContainer = {
  minHeight: '100%'
}
