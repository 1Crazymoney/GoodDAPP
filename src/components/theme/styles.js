import DefaultTheme from 'react-native-paper/src/styles/DefaultTheme'
import normalize from 'react-native-elements/src/helpers/normalizeText'

export const theme = {
  ...DefaultTheme,
  borders: {
    defaultBorderColor: '#c9c8c9',
  },
  colors: {
    ...DefaultTheme.colors,
    blue: '#006EA0',
    darkBlue: '#0C263D',
    darkGray: '#42454A',
    gray50Percent: '#CBCBCB',
    placeholder: '#CBCBCB',
    green: '#00C3AE',
    lightGray: '#EEE',
    lighterGray: '#A3A3A3',
    orange: '#F8AF40',
    primary: '#00AFFF',
    purple: '#9F6A9D',
    red: '#FA6C77',
    text: '#222',
  },
  fonts: {
    ...DefaultTheme.fonts,
    slab: 'RobotoSlab-Regular',
    slabBold: 'RobotoSlab-Bold',
    bold: 'Roboto-Bold',
  },
  paddings: {
    mainContainerPadding: normalize(8),
    defaultMargin: normalize(8),
  },
  sizes: {
    default: normalize(8),
    defaultDouble: normalize(16),
    defaultHalf: normalize(4),
  },
  fontStyle: {
    color: '#555',
    fontSize: normalize(18),
    textAlign: 'center',
  },
  modals: {
    overlayBackgroundColor: 'rgba(255, 255, 255, 0.8)',
    overlayHorizontalPadding: normalize(20),
    overlayVerticalPadding: normalize(30),
    backgroundColor: '#fff',
    jaggedEdgeSize: normalize(15),
    contentPadding: normalize(16),
    borderRadius: normalize(5),
    borderLeftWidth: normalize(10),
  },
  feedItems: {
    borderRadius: normalize(8),
    itemBackgroundColor: '#fff',
    height: normalize(84),
  },
}
