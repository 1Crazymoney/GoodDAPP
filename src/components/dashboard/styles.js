import normalize from 'react-native-elements/src/helpers/normalizeText'
import { StyleSheet } from 'react-native'

import { fontStyle } from '../common/styles'

export const receiveStyles = StyleSheet.create({
  wrapper: {
    justifyContent: 'flex-start',
    width: '100%',
    padding: normalize(8)
  },
  section: {
    flex: 1
  },
  sectionRow: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%'
  },
  qrCode: {
    marginTop: '2rem',
    padding: '1rem',
    borderColor: '#555',
    borderWidth: 1,
    borderRadius: '4px'
  },
  addressSection: {
    marginBottom: '1rem'
  },
  address: {
    margin: '0.5rem'
  },
  secondaryText: {
    margin: '1rem',
    color: '#555555',
    fontSize: normalize(14),
    textTransform: 'uppercase'
  },
  lowerSecondaryText: {
    margin: '1rem',
    color: '#555555',
    fontSize: normalize(16)
  },
  headline: {
    ...fontStyle,
    textTransform: 'uppercase',
    marginBottom: '1rem',
    fontSize: normalize(24)
  },
  buttonGroup: {
    width: '100%',
    flexDirection: 'row',
    marginTop: '1rem'
  },
  inputField: {
    width: '100%',
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  amountLabel: {
    ...fontStyle,
    fontSize: normalize(32)
  },
  amountSymbol: {
    fontSize: normalize(12)
  },
  amountWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignContent: 'center'
  },
  amountInput: {
    backgroundColor: 'transparent',
    height: normalize(40),
    width: '100%'
  },
  amountInputWrapper: {
    fontSize: normalize(26),
    lineHeight: normalize(40),
    whiteSpace: 'normal',
    flexShrink: 1,
    flexGrow: 1,
    textAlign: 'right'
  },
  shareQRButton: {
    width: '100%'
  },
  doneButton: {
    marginTop: '1rem'
  },
  fullWidth: {
    marginHorizontal: normalize(10)
  },
  amountSuffix: {
    flexGrow: 1,
    height: normalize(40),
    fontSize: normalize(10),
    justifyContent: 'center',
    lineHeight: normalize(40),
    paddingTop: normalize(10)
  },
  buttonStyle: {
    marginTop: '1em'
  }
})
