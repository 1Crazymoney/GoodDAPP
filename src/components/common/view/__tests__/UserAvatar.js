import React from 'react'
import renderer from 'react-test-renderer'
import { Provider as PaperProvider } from 'react-native-paper'
import UserAvatar from '../UserAvatar'
import { theme } from '../../../theme/styles'

// Note: test renderer must be required after react-native.

describe('UserAvatar', () => {
  const profile = {
    fullName: 'John Doe',
    email: 'a****a@aaa.com',
    mobile: '*********444'
  }

  it('renders without errors', () => {
    const tree = renderer.create(
      <PaperProvider theme={theme}>
        <UserAvatar profile={profile} />
      </PaperProvider>
    )
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <UserAvatar profile={profile} />
      </PaperProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
