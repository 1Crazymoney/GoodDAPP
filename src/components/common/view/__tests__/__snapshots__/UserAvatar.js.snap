// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxDirection": "normal",
        "WebkitBoxFlex": 1,
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "center",
        "WebkitFlexBasis": "0%",
        "WebkitFlexDirection": "row",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "WebkitJustifyContent": "center",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginTop": "50px",
        "msFlexDirection": "row",
        "msFlexNegative": 1,
        "msFlexPack": "center",
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "vertical",
          "WebkitFlexDirection": "column",
          "flexDirection": "column",
          "msFlexDirection": "column",
        }
      }
    >
      <div
        aria-disabled={true}
        className="css-view-1dbjc4n"
        disabled={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "MozTransitionProperty": "opacity",
            "MozUserSelect": "none",
            "WebkitTransitionDuration": "0.15s",
            "WebkitTransitionProperty": "opacity",
            "WebkitUserSelect": "none",
            "backgroundColor": "rgba(0,0,0,0.00)",
            "msUserSelect": "none",
            "transitionDuration": "0.15s",
            "transitionProperty": "opacity",
            "userSelect": "none",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "backgroundColor": "rgba(238,238,238,1.00)",
              "borderBottomLeftRadius": "60px",
              "borderBottomRightRadius": "60px",
              "borderTopLeftRadius": "60px",
              "borderTopRightRadius": "60px",
              "height": "120px",
              "width": "120px",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitFlexBasis": "auto",
                "borderBottomLeftRadius": "60px",
                "borderBottomRightRadius": "60px",
                "borderTopLeftRadius": "60px",
                "borderTopRightRadius": "60px",
                "flexBasis": "auto",
                "height": "120px",
                "msFlexPreferredSize": "auto",
                "overflowX": "hidden",
                "overflowY": "hidden",
                "width": "120px",
                "zIndex": 0,
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.00)",
                  "backgroundPosition": "center",
                  "backgroundRepeat": "no-repeat",
                  "backgroundSize": "cover",
                  "bottom": "0px",
                  "height": "100%",
                  "left": "0px",
                  "position": "absolute",
                  "right": "0px",
                  "top": "0px",
                  "width": "100%",
                  "zIndex": -1,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(85,85,85,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto",
            "fontSize": "24px",
            "fontWeight": "500",
            "lineHeight": "22px",
            "marginBottom": "8px",
            "marginTop": "8px",
            "textAlign": "center",
            "textDecoration": "none",
            "textTransform": "uppercase",
          }
        }
      >
        John Doe
      </div>
    </div>
  </div>
</div>
`;
